### ISSUE TITLE
 Too many certificates already issued for domain-a.com domain-b.com

### ISSUE THREAD

**Issue by sfinx26** [letsencrypt/letsencrypt#1816]
I have received 5 certificates, but when I try to get even, get the message.
information on existing limits can not find help to understand

**Comment by hrstoyanov** [comment]
This is a major issue. There must be some ```-renew-only-if-missing``` option for automation. The current behavior to ask question like "Do you want to (R)eplace or (C)ancel" will ruin your automated deployments. Specifying ```--renew-by-default``` will easily lead to this issue, even with modest number of (successful runs) wile testing your automated scripts.

**Comment by chriscroome** [comment]
@hrstoyanov the way I worked around this issue was to write a bash script which checks if a cert already exists for the `ServerName` and then checks if the `ServerAlias`' match the `subjectAltNames` in the existing cert and only if they don't does it then proceed with requesting a new one... I'm also very much looking forward to the 5 certs per TLD per week limit to be lifted...

**Comment by hrstoyanov** [comment]
@chriscroome I see (canyou share your scripts?), but I still think a ```--renew-only-if-needed``` option is the best solution. Something they should have thought about already prior to releasing... 
For example:
```
./letsencrypt-auto certonly  -a webroot --renew-only-if-needed --webroot-path /somewhere/webroot/ --email me@gmail.com --agree-tos -d mydomain.com
```

This way, one can test the deployment scripts at will without fearing of limits as no redundant certificates will be issued. As I mentioned, you can do:
```
./letsencrypt-auto certonly  -a webroot  --webroot-path /somewhere/webroot/ --email me@gmail.com --agree-tos -d mydomain.com
```
but your script will be interrupted with questions.

**Comment by chriscroome** [comment]
This is a ugly, but this is how I'm getting the `subjectAltName`'s from existing certs and munging them into something that can be compared to the `ServerAlias`' the Apache `VirtualHost` has...

```bash
CERT_SAN=$(openssl x509 -text -in /etc/letsencrypt/live/$SITENAME/cert.pem -noout | \
         grep "DNS" | tr -d "DNS:" | sed 's/\ /\n/g' | sort |  tr  "\n" " " | sed 's/^\s\+//' | \
         sed 's/,\ $//' | sed 's/,//g' | sed 's/ /,/g' | sed 's/,$//' )
```

Clearly having something like a `--renew-only-if-needed` option would be better ;-)

**Comment by sfinx26** [comment]
hm i use  ./letsencrypt-auto certonly --standalone --renew-only-if-needed --email mail.mail.com -d domaina.com
and iam rly confused 

letsencrypt: error: unrecognized arguments: --renew-only-if-needed

**Comment by pde** [comment]
@hrstoyanov that ticket is #1546

**Comment by pde** [comment]
@sfinx26 you were reading a proposal for a flag that could be implemented, not one that already exists.

**Comment by hrstoyanov** [comment]
@pde thanks for referencing #1546, it is the same issue. Do you guys have a Super-Urgent label :-)  ?

**Comment by sfinx26** [comment]
Hm...
and now i get this error for request NEW domain (this domain had no previous certificates/or cretificates request)

**Comment by patf** [comment]
@sfinx26 crt.sh shows 5 certificates for subdomains of prodiplom.net issued on 2015-12-07:
https://crt.sh/?q=prodiplom%2Enet&iCAID=7395

Rate limits are applied on a TLD+1 level, i.e. example.com, www.example.com, foo.example.com all count towards the rate limit of example.com.

BTW, the community forum is a better place for issues like these.

**Comment by pde** [comment]
#1546 has landed in master and should be in the 0.1.1 release on Tuesday.  It removes a significant cause of users repeatedly making certs, and also presents information about rate limits in the exact menu option that users select when renewing early.
